regimes/mx/complements/fuel_complement.go

```diff
@@ -0,0 +1,58 @@
+// Package complements provides GOBL Complements for Mexican invoices
+package complements
+
+import (
+	"cloud.google.com/go/civil"
+	"github.com/invopop/gobl/bill"
+	"github.com/invopop/gobl/cbc"
+	"github.com/invopop/gobl/num"
+	"github.com/invopop/gobl/org"
+	"github.com/invopop/gobl/tax"
+)
+
+// FuelComplement provides detailed information about fuel purchases made with
+// electronic wallets. In Mexico, e-wallet suppliers are required to report this
+// information as part of the invoices they issue to their customers.
+type FuelComplement struct {
+	// Customer's e-wallet account number
+	WalletAccountNumber string `json:"wallet-account-number" jsonschema:"title=Wallet Account Number"`
+	// List of fuel purchases made with the customer's e-wallets
+	Lines []*FuelLine `json:"lines" jsonschema:"title=Lines"`
+	// Summary of all the purchases totals, including taxes (calculated)
+	Totals bill.Totals `json:"totals" jsonschema:"title=Totals" jsonschema_extras:"calculated=true"`
+}
+
+// FuelLine represents a single fuel purchase made with an e-wallet issued by
+// the invoice's supplier to the invoice's customer.
+type FuelLine struct {
+	// Identifier of the e-wallet used to make the purchase
+	WalletID cbc.Code `json:"wallet-id" jsonschema:"title=Wallet Identifier"`
+	// Date and time of the purchase
+	PurchaseDateTime civil.DateTime `json:"purchase-date" jsonschema:"title=Purchase Date"`
+	// Tax ID (RFC) of the purchaser
+	PurchaserTaxID cbc.Code `json:"purchaser-tax-id" jsonschema:"title=Purchaser's Tax ID"`
+	// Code of the service station where the purchase was made
+	ServiceStationCode cbc.Code `json:"service-station-code" jsonschema:"title=Service Station Code"`
+	// Amount of fuel units purchased
+	Quantity num.Amount `json:"quantity" jsonschema:"title=Quantity"`
+	// Details about the fuel purchased
+	Item FuelItem `json:"item" jsonschema:"title=Item"`
+	// Identifier of the purchase (folio)
+	PurchaseID cbc.Code `json:"purchase-id" jsonschema:"title=Purchase Identifier"`
+	// Result of quantity multiplied by the item's price (calculated)
+	Sum num.Amount `json:"sum" jsonschema:"title=Sum" jsonschema_extras:"calculated=true"`
+	// Map of taxes applied to the purchase
+	Taxes tax.Set `json:"taxes" jsonschema:"title=Taxes"`
+}
+
+// FuelItem provides the details of the fuel purchased.
+type FuelItem struct {
```

Comment:
As discussed in other parts of this PR, the similarity of this struct with `org.Item` makes it tempting to reusing. In this particular case, I decided not to, but the considerations discussed in other comments could be applied here to make a different decision. WDYT?

Reply:
I'd be tempted to store the `FuelItem` fields inside the `FuelLine`... the criteria is dependent on how re-usable the Item needs to be. Will the `FuelItem` ever need to be persisted independently?

Reply:
I created a separate `FuelItem` struct just to mirror the existing pattern in `bill.Line` and `org.Item`. It neatly separates item details from purchase info, and it's an idiom that GOBL developers will be familiar with. I also expect this data to be persisted separately in the source systems.

On the other hand, in the CFDI the line and item data are at the same level. So if you're not convinced by this approach, let me know and I'll flatten down here too.

Reply:
My thoughts are around the path of least resistance for users trying to figure out what to do. I think it might be a good idea in this case to stick as closely as possible to the original source structure, with translated fields...

Reply:
> My thoughts are around the path of least resistance for users trying to figure out what to do.

Reply:
I think I accidentally deleted your last comment here. Sorry for the mess, I'm bringing it back:

@samlown said:
> My thoughts are around the path of least resistance for users trying to figure out what to do. I think it might be a good idea in this case to stick as closely as possible to the original source structure, with translated fields...

Finding the path of least resistance for users is the same design principle I'm trying to follow. Question is which path is that! From my perspective, one of the main benefits that GOBL brings to developers is providing them with a single unified language, with a consistent vocabulary and set of idioms and patterns that they can use across regimes. That's why I thought the least resistance path was modelling lines and items resembling the ones in GOBL.

In any case, I'll propose an alternative version sticking as closely as possible to the original source structure with translated fields.